bugbug/models/rcatype.py

```diff
@@ -0,0 +1,193 @@
+
+        self.RCA_LIST = sorted(set(self.RCA_TYPES))
+
+        feature_extractors = [
+            bug_features.has_str(),
+            bug_features.severity(),
+            # Ignore keywords that would make the ML completely skewed
+            # (we are going to use them as 100% rules in the evaluation phase).
+            bug_features.is_coverity_issue(),
+            bug_features.has_crash_signature(),
+            bug_features.has_url(),
+            bug_features.has_w3c_url(),
+            bug_features.has_github_url(),
+            # bug_features.whiteboard(),
+            bug_features.patches(),
+            bug_features.landings(),
+            bug_features.blocked_bugs_number(),
+            bug_features.ever_affected(),
+            bug_features.affected_then_unaffected(),
+            bug_features.product(),
+            bug_features.component(),
+        ]
+
+        cleanup_functions = [
+            feature_cleanup.url(),
+            feature_cleanup.fileref(),
+            feature_cleanup.synonyms(),
+        ]
+
+        self.extraction_pipeline = Pipeline(
+            [
+                (
+                    "bug_extractor",
+                    bug_features.BugExtractor(feature_extractors, cleanup_functions),
+                ),
+                (
+                    "union",
+                    ColumnTransformer(
+                        [
+                            ("data", DictVectorizer(), "data"),
+                            ("title", self.text_vectorizer(min_df=0.001), "title"),
+                            (
+                                "first_comment",
+                                self.text_vectorizer(min_df=0.001),
+                                "first_comment",
+                            ),
+                            (
+                                "comments",
+                                self.text_vectorizer(min_df=0.001),
+                                "comments",
+                            ),
+                        ]
+                    ),
+                ),
+            ]
+        )
+
+        self.clf = OneVsRestClassifier(xgboost.XGBClassifier(n_jobs=16))
+
+    def get_rca(self, bugs):
+        all_rca = []
+        bug_ids = []
+        for bug in bugs:
+            bug_ids.append(int(bug["id"]))
+            rca = []
+            whiteboard_data = re.sub(" ", "", bug["whiteboard"]).lower()
+            for whiteboard in whiteboard_data.split("["):
+                delimiter = (
+                    whiteboard[3]
+                    if (whiteboard.startswith("rca-") or whiteboard.startswith("rca:"))
+                    else False
+                )
+                if delimiter:
+                    rca_whiteboard = re.sub("]", "", whiteboard)
+                    rca_whiteboard_split = (
+                        rca_whiteboard.split(delimiter, 1)
+                        if self.rca_subcategories_enabled
+                        else rca_whiteboard.split(delimiter)
```

Comment:
How does this work when you have something like: `rca: codingerror - compatibilityissue`? The delimiter you get earlier is `:`, but you should split by `-`.
(This is a good reason to have some unit tests like mentioned above)

Reply:
Oh, I didn't know we could have such "hybrid" cases too. Will add support for these types also.